### ISSUE TITLE
invalid for-each argument in sa.tf

### ISSUE THREAD

**Issue by ananthmkh-gcp** [terraform-google-modules/terraform-google-kubernetes-engine#1605]
### TL;DR

in sa.tf line #72 and # 85 return error mentioning "The "for_each" set includes values derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will identify the instances of this resource."

### Observed behavior

Error Snippet:
 Error: Invalid for_each argument
│ 
│   on ../modules/autopilot-public-cluster/sa.tf line 75, in resource "google_project_iam_member" "cluster_service_account-gcr":
│   75:   for_each = var.create_service_account && var.grant_registry_access ? toset(local.registry_projects_list) : []
│     ├────────────────
│     │ local.registry_projects_list is list of string with 1 element
│     │ var.create_service_account is true
│     │ var.grant_registry_access is true
│ 
│ The "for_each" set includes values derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine
│ the full set of keys that will identify the instances of this resource.
│ 
│ When working with unknown values in for_each, it's better to use a map value where the keys are defined statically in your configuration and
│ where only the values contain apply-time results.
│ 
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then
│ apply a second time to fully converge.
╵
╷
│ Error: Invalid for_each argument
│ 
│   on ../modules/autopilot-public-cluster/sa.tf line 82, in resource "google_project_iam_member" "cluster_service_account-artifact-registry":
│   82:   for_each = var.create_service_account && var.grant_registry_access ? toset(local.registry_projects_list) : []
│     ├────────────────
│     │ local.registry_projects_list is list of string with 1 element
│     │ var.create_service_account is true
│     │ var.grant_registry_access is true
│ 
│ The "for_each" set includes values derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine
│ the full set of keys that will identify the instances of this resource.
│ 
│ When working with unknown values in for_each, it's better to use a map value where the keys are defined statically in your configuration and
│ where only the values contain apply-time results.
│ 
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then
│ apply a second time to fully converge.

### Terraform Configuration

```hcl
module "gkeapp" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-autopilot-public-cluster"
  project_id                 = module.envprj.project_id
  name                       =  "${var.gkeapp_cluster_name}"
  region                     = local.region
  zones                      = local.zones

  kubernetes_version         = var.gkeapp_master_version
  release_channel            = var.gkeapp_release_channel


  service_account = module.sa_gkeapp_node.email
  grant_registry_access = true

  network                    = module.vpc.network_name
  subnetwork                 = "subnet-app"
  ip_range_pods              = "secondary-range-app-subnet-gke-pods"
  ip_range_services          = "secondary-range-app-subnet-gke-services"

  master_authorized_networks = var.gkeapp_master_authorized_networks 

  maintenance_start_time = var.maintenance_start_time_gkeapp_k8s
  maintenance_end_time = var.maintenance_end_time_gkeapp_k8s
  maintenance_exclusions = var.maintenance_gkeapp_exclusion

  horizontal_pod_autoscaling = true
}
```

### Terraform Version

```sh
terraform version
Terraform v1.4.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.60.1
+ provider registry.terraform.io/hashicorp/google-beta v4.60.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.19.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1

Your version of Terraform is out of date! The latest version
is 1.4.4. You can update by downloading from https://www.terraform.io/downloads.html
```
